cardano-cli/src/Cardano/CLI/EraBased/Run/Governance/DRep.hs

```diff
@@ -150,10 +170,25 @@ runGovernanceDRepUpdateCertificateCmd
     , outFile
     } =
     conwayEraOnwardsConstraints w $ do
+      mapM_
+        (hashCheckToGovernanceCmdError . carryHashChecks)
+        mAnchor
       drepCredential <- modifyError GovernanceCmdKeyReadError $ readDRepCredential drepHashSource
-      let updateCertificate = makeDrepUpdateCertificate (DRepUpdateRequirements w drepCredential) mAnchor
+      let updateCertificate =
+            makeDrepUpdateCertificate
+              (DRepUpdateRequirements w drepCredential)
+              (pcaAnchor <$> mAnchor)
       firstExceptT GovernanceCmdTextEnvWriteError . newExceptT $
         writeFileTextEnvelope outFile (Just "DRep Update Certificate") updateCertificate
+   where
+    hashCheckToGovernanceCmdError :: ExceptT HashCheckError IO () -> ExceptT GovernanceCmdError IO ()
```

Comment:
Why not add a new constructor to `GovernanceCmdError` to encapsulate `HashCheckError`?  This function has been repeatedly defined.